lib/slack/lookups.ex

```diff
@@ -28,13 +28,15 @@ defmodule Slack.Lookups do
 
   @doc ~S"""
   Turns a string like `"#CHANNEL_NAME"` into the ID that Slack understands
-  (`"C…"`).
+  (`"C…"`) if a public channel,
+  (`"G…"`) if a group/private channel.
   """
   def lookup_channel_id("#" <> channel_name, slack) do
-    slack.channels
-    |> Map.values
-    |> Enum.find(fn channel -> channel.name == channel_name end)
-    |> Map.get(:id)
+    channel_finder = fn {_id, map} -> if map.name == channel_name, do: map, else: nil end
```

Comment:
Can this be extracted into a helper function? Maybe include the `Enum.find_value`?

`defp find_channel_by_name` maybe?